typing/printtyp.ml

```diff
@@ -1581,6 +1578,10 @@ type rec_item_group =
   | Not_rec of syntactic_sig_item
   | Rec_group of (Ident.t list * syntactic_sig_item list)
 
+let with_hidden_items ~is_type ids f =
+  if is_type && not !Clflags.real_paths then
+    wrap_env (hide ids) (Naming_context.with_hidden ids f)
+  else Naming_context.with_hidden ids f
```

Comment:
The code of this function is short, but it is a tricky piece of code. I wish you would give a name to the first wrapper, for example `with_hidden_in_printing_env`, and avoid the duplication, so that the structure is clear.

Reply:
I am not completely convinced, but why not?